packages/mesh-graphql-client/src/index.ts

```diff
@@ -0,0 +1,372 @@
+            expirationTimeSeconds
+            salt
+            signature
+            fillableTakerAssetAmount
+        }
+    }
+`;
+
+const orderEventsSubscription = gql`
+    subscription {
+        orderEvents {
+            timestamp
+            endState
+            order {
+                hash
+                chainId
+                exchangeAddress
+                makerAddress
+                makerAssetData
+                makerAssetAmount
+                makerFeeAssetData
+                makerFee
+                takerAddress
+                takerAssetData
+                takerAssetAmount
+                takerFeeAssetData
+                takerFee
+                senderAddress
+                feeRecipientAddress
+                expirationTimeSeconds
+                salt
+                signature
+                fillableTakerAssetAmount
+            }
+            contractEvents {
+                blockHash
+                txHash
+                txIndex
+                logIndex
+                isRemoved
+                address
+                kind
+                parameters
+            }
+        }
+    }
+`;
+
+export class MeshGraphQLClient {
+    // private readonly _subscriptionClient: SubscriptionClient;
+    private readonly _client: ApolloClient<NormalizedCacheObject>;
+    constructor(httpUrl: string, webSocketUrl: string) {
+        // Set up an apollo client with WebSocket and HTTP links. This allows
+        // us to use the appropriate transport based on the type of the query.
+        const httpLink = new HttpLink({
+            uri: httpUrl,
+        });
+        const wsSubClient = new SubscriptionClient(
+            webSocketUrl,
+            {
+                reconnectionAttempts: 3,
+                reconnect: true,
+                connectionCallback: err => {
+                    if (err) {
+                        // console.error(err);
+                    } else {
+                        // console.log('successfully connected');
+                    }
+                },
+            },
+            // Use ws in Node.js and native WebSocket in browsers.
+            (process as any).browser ? undefined : ws,
+        );
+        // wsSubClient.onError(err => console.error(err));
+        // wsSubClient.onDisconnected(() => console.error('detected disconnect'));
+        const wsLink = new WebSocketLink(wsSubClient);
+        const splitLink = split(
+            ({ query }) => {
+                const definition = getMainDefinition(query);
+                return definition.kind === 'OperationDefinition' && definition.operation === 'subscription';
+            },
+            wsLink,
+            httpLink,
+        );
+        const errorLink = onError(({ graphQLErrors, networkError }) => {
+            if (graphQLErrors != null && graphQLErrors.length > 0) {
+                // Throw the first error.
+                // TODO(albrow): Is there a clean way to include all the errors?
```

Comment:
Is there any reason why we couldn't just create a table with the format that you use below for the first error?

Reply:
A table? What do you mean by that in the context of a JavaScript error?

The key word here is "clean". We can just append all the messages together in a single string but I was trying to think if there was a better way to do it. Any ideas?

Reply:
I was thinking of creating a "table" by appending the messages together in a single string with new-lines. This may not be clean in practice, but I think that would be better if it formats well in the browser.

Reply:
Hmm.. I'm not sure how I feel about this. I generally don't put newlines in error messages and in the Go world this isn't really a thing. IMO error messages should be short and concise. Newlines render nicely in some contexts (e.g. logging the error message and nothing else with `console.log` or `console.error`) but render worse in others (e.g., in our logging infrastructure, the error is part of a map/JSON object so it would render as `\n` not an actual line break).